pkg/reconciler/serverlessservice/resources/services.go

```diff
@@ -98,14 +108,18 @@ func filterSubsetPorts(targetPort int32, subsets []corev1.EndpointSubset) []core
 	}
 	ret := make([]corev1.EndpointSubset, len(subsets))
 	for i, sss := range subsets {
+		org := sss.DeepCopy()
 		sst := sss.DeepCopy()
 		// Find the port we care about and remove all others.
-		for j, p := range sst.Ports {
+		for j, p := range org.Ports {
```

Comment:
what if we avoid the deep copy for `org` and we do `for j, p := range sss.Ports`?

Reply:
Yes, there are several reasons:
- non-admin users cannot notice if they are using secure traffic or not.
- It is easier to debug as you said.
- It makes easy to use Kingress by extra components. (OpenShift uses an extra controller to creates a LB resource based on the Kingress.)